fvm/programs/block_derived_data.go

```diff
@@ -7,6 +7,10 @@ import (
 	"github.com/onflow/flow-go/fvm/state"
 )
 
+type ValueComputer[TKey any, TVal any] interface {
+	Compute(txnState *state.TransactionState, key TKey) (TVal, error)
```

Comment:
Can you add some comments to this method?

Is it to get value from the state under the given key? Why it's not called `GetValue`? What is the actual "Computation"?

Reply:
Could you also explain why do we need to use those generic type `TKey` and `TVal`?

Reply:
(will add comment as suggested)

This is used by BlockDerivedData, which is a schema-ed ACI (not D) database table for stroing TKey/TVal pairs.  This is generic because BlockDerivedData is generic.